### PR TITLE
Add needle in yaml configurations files

### PR DESCRIPTION
**Pull request by moifort**
Allow to add/modify config.yml (main, prod and dev), but it's delete all comments inside yaml file.

### CODE REVIEW

File: script-base.js [hunk 1]
```diff
@@ -126,6 +126,35 @@ Generator.prototype.addElementTranslationKey = function(key, value, language) {
...
+ *  - 'main' -> application.yml (default)
+ *  - 'dev' -> application-dev.yml
+ *  - 'prod' -> application-prod.yml
+ */
+Generator.prototype.updateApplicationConfiguration = function(configObj, file) {
+    var fullPath;
+    if (file === 'dev') {
+        fullPath = 'src/main/resources/config/application-dev.yml';
+    }
+    else if (file === 'prod') {
```
Comment (pascalgrimaud) [inline script-base.js]: Format: `} else if ...`

File: script-base.js [hunk 2]
```diff
@@ -126,6 +126,35 @@ Generator.prototype.addElementTranslationKey = function(key, value, language) {
...
+        fullPath = 'src/main/resources/config/application-prod.yml';
+    }
+    else {
```
Comment (pascalgrimaud) [inline script-base.js]: Format: `} else if ...`

File: script-base.js [hunk 3]
```diff
@@ -126,6 +126,35 @@ Generator.prototype.addElementTranslationKey = function(key, value, language) {
...
+        fullPath = 'src/main/resources/config/application.yml';
+    }
+    console.log(chalk.yellow('\nupdate ') + fullPath);
+    try {
+        jhipsterUtils.rewriteYAMLFile(fullPath, configObj);
+    } catch (e) {
+        console.log(chalk.yellow('\nUnable to find ') + fullPath + chalk.yellow('. Reference to ') + 'bower dependency (name: ' + e + ')' + chalk.yellow(' not added.\n'));
```
Comment (pascalgrimaud) [inline script-base.js]: Need to fix the message, there is no bower dependency here :smile:

**Comment by moifort** [comment]
Oups, what happens when you make a PR 2am.

**Comment by jdubois** [comment]
Not a big fan of rewriting the Yaml file and losing all the comments... And probably also the needles in it

Can't you do this with a needle? What are you trying to do?

**Comment by moifort** [comment]
I want to add a new client Id and secret id for a new social connection. I start with needle, but after that, I think it would be better to make the same like bower (adding dependency). I will make the refacto. 

Envoyé de mon iPhone

**Comment by moifort** [comment]
@jdubois ca te convient?

**Comment by cbornet** [comment]
Can't we make it more generic so that we can add any conf ?

**Comment by moifort** [comment]
I think the only way to make it generic it was my work below that rebuild a yml file. But the problem is that delete all the comment inside the yml config. That's why I refactor this function to add only social config needle. For today YAGNI if we need change after we will change it :)



Envoyé de mon iPhone

**Comment by cbornet** [comment]
@moifort we could have a needle at the end of the file that allows to add any conf. It just would not be well classified with the existing confs.

**Pull request merged**